solutions/instances/main.tf

```diff
@@ -242,6 +301,22 @@ module "observability_instance" {
   activity_tracker_routes = local.at_routes
 }
 
+resource "time_sleep" "wait_for_atracker_cos_authorization_policy" {
+  count           = var.ibmcloud_cos_api_key == null ? 0 : 1
+  depends_on      = [ibm_iam_authorization_policy.atracker_cos]
+  create_duration = "30s"
+}
+resource "ibm_iam_authorization_policy" "atracker_cos" {
+  count                       = var.ibmcloud_cos_api_key != null && !var.skip_at_cos_auth_policy ? 1 : 0
+  provider                    = ibm.cos
+  source_service_account      = data.ibm_iam_account_settings.iam_account_settings.account_id
+  source_service_name         = "atracker"
+  target_service_name         = "cloud-object-storage"
+  target_resource_instance_id = regex(".*:(.*)::", local.cos_instance_crn)[0]
+  roles                       = ["Object Writer"]
+  description                 = "Permit AT service Object Writer access to COS instance ${local.cos_instance_crn}"
+}
```

Comment:
I understand this code allows AT events routing to cos bucket. Any specific reason that auth policy is for cos instance and not for bucket?

Reply:
@maheshwarishikha  Before creating ibm_iam_authorization_policy here, we was only passing the boolean variable `skip_atracker_cos_iam_auth_policy` and auth policy is getting created using root module https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/blob/6b88ba2e51980bc291581f62f62111788fcf38bf/modules/activity_tracker/main.tf#L11 .
there also only auth policy is getting created for instance.

Reply:
hmm, ok.
It is done in the same way as observability-instances module does. So for now, its ok. We can revisit this if required.